### ISSUE TITLE
Font size too small in Chinese characters

### ISSUE THREAD

**Issue by SennatSu** [desktop/desktop#3543]
Can you adjust "--font-size-sm: 11px" to "--font-size-sm: 12px" ?

We found that if our code include Chinese characters, the display effect will be very bad.

You can see my screenshots below, each time we use github desktop, we must adjust it by ourselves, otherwise we cannot see our codes clearly.

"--font-size-sm: 11px" (Default)
![001](https://user-images.githubusercontent.com/24712594/33761388-cc949578-dc43-11e7-9456-bc0ad8361de0.png)
![002](https://user-images.githubusercontent.com/24712594/33761431-f29e548e-dc43-11e7-9637-7a7c16948b26.png)

"--font-size-sm: 12px" (Optimized)
![003](https://user-images.githubusercontent.com/24712594/33761459-078022f6-dc44-11e7-85b2-84e4060196e0.png)
![004](https://user-images.githubusercontent.com/24712594/33761464-0b44e6e2-dc44-11e7-9c63-21ff9db81b9d.png)

We really hope you can think about it, thanks!

**Comment by say25** [comment]
Any reason why zoom doesn't fix this? `View -> Zoom In`

**Comment by iAmWillShepherd** [comment]
Thank you @SennatSu for taking the time to detail out this issue 👍. I'll let @desktop/design weigh in on this.

**Comment by SennatSu** [comment]
@say25 Thank you man, "Zoom In" can fix my problem, but I just don't want to change the size of the frame~

**Comment by kasyanov-ms** [comment]
Also "Zoom" on non-Retina displays makes some elements blurred
![blur](https://user-images.githubusercontent.com/21953011/34071461-b859e3b6-e287-11e7-9066-40400975c01a.png)

**Comment by kasyanov-ms** [comment]
Also "Zoom" function on non-Retina displays makes some elements blurred
![blur](https://user-images.githubusercontent.com/21953011/34071490-02c9153e-e288-11e7-9e7a-af44cecdb37c.png)
I believe that the monospace font should match the GitHub.com diffs

**Comment by hoxily** [comment]
## Version

* GitHub Desktop:1.2.2
* Operating system:Microsoft Windows [版本 10.0.15063]
* Operating system language：zh_CN，简体中文

The default zoom level displays files changes badly when there are Chinese characters. Here is my screenshot below:

![default](https://user-images.githubusercontent.com/1575029/40756403-07d7ffc6-64b6-11e8-9000-5cb6cc3c79b2.png)
(**you need click the picture to make it display in real size. otherwise it's hard to inspect.**)


The next zoom level dispays Chinese characters clearly, provided below:

![default](https://user-images.githubusercontent.com/1575029/40756551-d08fdaf6-64b6-11e8-935f-a8bbf384c16f.png)
(**you need click the picture to make it display in real size. otherwise it's hard to inspect.**)

## Expected Behavior

Hope the default zoom level is set to which the second screeshot have.

**Comment by hoxily** [comment]
I find that I can change `--font-size-sm` from 11px to 12px to solve this problem.
The configure css file is located at C:\Users\Administrator\AppData\Local\GitHubDesktop\app-1.2.2\resources\app\renderer.css  in `:root` section.
Hope this helps you.

**Comment by SennatSu** [comment]
Thank you all.

**Comment by shiftkey** [comment]
Reopening this to track whether there is other work that we should do in here.

**Comment by kasyanov-ms** [comment]
> I find that I can change `--font-size-sm` from 11px to 12px to solve this problem.
> The configure css file is located at C:\Users\Administrator\AppData\Local\GitHubDesktop\app-1.2.2\resources\app\renderer.css in `:root` section.
> Hope this helps you.
Unfortunately, doesn't work on 1.5 version

**Comment by jorgelbg** [comment]
I think that in general, the desktop app should mimic the font size of the web UI.

**Comment by SylwesterZarebski** [comment]
It should be possible to change not only font size, but also family with weight, eg. to change code font to Fira Code or Inconsolata (with desired weight and size of course).

**Comment by caop3ng** [comment]
How can I only change code font size at most right?

**Comment by Milad101** [comment]
Changing the zoom doesn't solve the problem when you're dealing with large code files. (e.g more than 500 lines of codes) besides the font being used is so bad that when you zoom in a couple of times sidebar, menus will get larger too and still, the changed file is not easy to read. The ability to not only change the font size but also changing the font itself for the preview of the changed files is a need.

**Comment by silvinor** [comment]
It's astounding how such a simple thing can procrastinate for so long. A simple preference on the preferences pane to choose a Font Family and Font Size (like Atom does!) is a easy solution to implement the countless issues reported on "font". (#9541, #9389, #5315, #4967, #3923, #3543, #10955, #10436, #9222, #8798, ... and on and on).

**Comment by steveward** [comment]
Requested in https://github.com/desktop/desktop/issues/15216.

**Comment by anthumchris** [comment]
@steveward It's almost time to wish this issue a happy 5th birthday 🎂 🎈

Given that zoom in/out currently have min/max boundaries, could you point me to the function that explicitly limits that range? I'm assuming I can submit a simple PR and quickly resolve https://github.com/desktop/desktop/issues/15216

**Comment by umarmughal** [comment]
> I find that I can change `--font-size-sm` from 11px to 12px to solve this problem. The configure css file is located at C:\Users\Administrator\AppData\Local\GitHubDesktop\app-1.2.2\resources\app\renderer.css in `:root` section. Hope this helps you.

@hoxily it's not working for me Windows 11 64bit, any other way please?

**Comment by xPaw** [comment]
I agree 11px is too small. Subtitles in pull request menu combined with dark gray color makes it quite hard to read:

![image](https://user-images.githubusercontent.com/613331/200892809-6f1349c6-1173-4f19-bb97-04784731fc9a.png)

**Comment by anthumchris** [comment]
Could we let users decide their fonts instead of us deciding for them? if the app can scale down to a non-fixed minimum font size, the usability goal is met.

**Comment by heathermh** [comment]
I would appreciate settings to select my own fonts and sizes. This is an accessibility issue.

**Comment by emmmarosewalker** [comment]
Agree that this is an easily fixed accessibility issue

**Comment by pravin** [comment]
Any updates on this?

**Comment by Jaeiya** [comment]
I'm starting to wonder if `font-size` is one of the hardest things to implement, seeing as not only this application, but also https://github.com/microsoft/vscode/issues/519 are taking multiple years to figure this out. Did something happen to make this suddenly a difficult problem to solve? Is it because of Electron?

**Comment by willser** [comment]
Anything new to this issue?:grinning:

**Comment by YahyaMohamedYahya** [comment]
Thank you

**Comment by silvinor** [comment]
If you're on Mac you can hide this in the CSS. Open and edit (with "Show Package Contents") `/Contents/Resources/app/renderer.css` and add this to the very bottom:

```css
.diff-code-mirror .CodeMirror,
.side-by-side-diff-container .side-by-side-diff {
  font-family: "Cartograph CF", SFMono-Regular, Consolas, "Liberation Mono", monospace !important;
  font-size: 16px !important;
  font-weight: 200 !important;
}
```

*Obviously chose your own font-family et al.*

*(Will need to be done each time it updates itself.)*

**Comment by kingdo9** [comment]
I've submitted a PR , hope it gets reviewed as I don't have a issue I can tag to it with Help Wanted label 
@sergiou87

**Comment by niik** [comment]
Hi folks, we discussed this issue in our most recent issue review meeting and the team has decided against implementing custom fonts and font sizes. We understand that this is disappointing for some of you and we hope you'll be able to leverage interface scaling (i.e. zoom) as a way to increase the font size across the app.

**Comment by anthumchris** [comment]
> Hi folks, we discussed this issue in our most recent issue review meeting and the team has decided against implementing custom fonts and font sizes.

This is still an Electron web app using CSS? Can you allow us to use custom CSS or an external stylesheets?
